lib/npm-installation-manager.ts

```diff
@@ -42,19 +42,22 @@ export class NpmInstallationManager implements INpmInstallationManager {
 		return path.join(this.getCacheRootPath(), packageName, version, "package");
 	}
 
-	public addToCache(packageName: string, version: string): IFuture<void> {
+	public addToCache(packageName: string, version: string): IFuture<any> {
 		return (() => {
 			let cachedPackagePath = this.getCachedPackagePath(packageName, version);
+			let cachedPackageData: any;
 			if(!this.$fs.exists(cachedPackagePath).wait() || !this.$fs.exists(path.join(cachedPackagePath, "framework")).wait()) {
-				this.addToCacheCore(packageName, version).wait();
+				cachedPackageData = this.addToCacheCore(packageName, version).wait();
 			}
-
-			if(!this.isShasumOfPackageCorrect(packageName, version).wait()) {
+			let realVersion = (cachedPackageData && cachedPackageData.version) || version;
+			if(!this.isShasumOfPackageCorrect(packageName, realVersion).wait()) {
 				// In some cases the package is not fully downloaded and there are missing directories
 				// Try removing the old package and add the real one to cache again
-				this.addCleanCopyToCache(packageName, version).wait();
+				cachedPackageData = this.addCleanCopyToCache(packageName, version).wait();
```

Comment:
According to the interface, `addCleanCopyToCache(packageName: string, version: string): IFuture<void>;`, so this reassignment of cachedPackageData is pointless.
I'm guessing the interface is a bit off.